pkg/cmd/server/origin/master.go

```diff
@@ -273,6 +271,55 @@ func initAPIVersionRoute(root *restful.WebService, version string) {
 		Consumes(restful.MIME_JSON))
 }
 
+// If we know the location of the asset server, redirect to it when / is requested
+// and the Accept header supports text/html
+func assetServerRedirect(handler http.Handler, assetPublicURL string) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, req *http.Request) {
+		accept := req.Header.Get("Accept")
+		if req.URL.Path == "/" && strings.Contains(accept, "text/html") {
```

Comment:
We're aware this isn't fully spec compliant. Feel free to add a todo or open a pull request, but it's working well enough for now (browsers get redirected, curl and requests that don't accept text/html explicitly get the JSON.